installers/msi-language/StateDeploy/CustomAction.cs

```diff
@@ -517,43 +542,99 @@ private static string BuildDeployCmd(Session session, string subCommand)
         }
 
         /* The following custom actions are added to this project (and not to a project
-         * with a more appropriate name) in hope that the TrackerSingleton ca be re-used between 
+         * with a more appropriate name) in hope that the TrackerSingleton ca be re-used between
          * all custom actions.
          */
 
         [CustomAction]
         public static ActionResult SetSessionID(Session session)
         {
-            session["SESSION_ID"] = Guid.NewGuid().ToString();
+            if (session["SESSION_ID"] != "unset") {
+                var msg = "In SetSessionID, but SESSION_ID is already set";
+                session.Log(msg);
+                RollbarReport.Error(msg, session);
+            }
+            session["SESSION_ID"] = NewSessionID(session, session["ProductVersion"]);
             return ActionResult.Success;
         }
 
         [CustomAction]
         public static ActionResult GAReportFailure(Session session)
         {
             session.Log("sending event about MSI failure");
-            TrackerSingleton.Instance.TrackEventSynchronously(session, session["SESSION_ID"], "stage", "finished", "failure", session["ProductVersion"]);
+            var sessionID = GetSessionIDForExitAction(session);
+            if (sessionID == "unset")
+            {
+                // this can happen if an installation error happens before we could initialize the session id and send the start event
+
+                // So, we create a new session id,
+                sessionID = NewSessionID(session, session["ProductVersion"]);
```

Comment:
It appears that `NewSessionID` now sends the started event but here it's being called from `GAReportFailure`. It's also called from `SetSessionID` which is scheduled in the `InstallUISequence` so I think this could lead to double reporting of the started event. 

It may be better to separate the two and specifically schedule both the `NewSessionID` and `SendStartedEvent` in the to be more explicit

Reply:
The idea was that when a new session id *must* be created, because it didn't exist before (was "unset"), that a start event cannot have been send yet.

But again that is obsolete now.